### PR TITLE
Created separate backup test files based on area of testing:PA-566

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
This PR splits the backup_test file into different files based on the area of testing
**Which issue(s) this PR fixes** (optional)
Closes #
https://portworx.atlassian.net/browse/PA-566
**Special notes for your reviewer**:

### CODE REVIEW

**Comment by mkoppal-px** [comment]
I think we can rename `tests/backup/backup_negative_test.go` to `tests/backup/backup_resiliency_test.go`

File: tests/backup/backup_helper.go [hunk 1]
```diff
@@ -27,7 +27,6 @@ const (
 	cloudAccountDeleteTimeout                 = 10 * time.Minute
 	cloudAccountDeleteRetryTime               = 30 * time.Second
 	storkDeploymentNamespace                  = "kube-system"
-	clusterName                               = "tp-cluster"
```
Comment (mkoppal-px) [inline tests/backup/backup_helper.go:30]: Was this removed because it was not being used by function?

Comment (snigdha-px) [reply tests/backup/backup_helper.go:30]: yes

Comment (snigdha-px) [reply tests/backup/backup_helper.go:30]: done

**Comment by snigdha-px** [comment]
> I think we can rename `tests/backup/backup_negative_test.go` to `tests/backup/backup_resiliency_test.go`

done

**Review by lsrinivas-pure — CHANGES_REQUESTED** [review]

File: tests/backup/backup_locked_bucket_test.go [hunk 2]
```diff
@@ -0,0 +1,407 @@
...
+	CloudCredUIDMap := make(map[string]string)
+	BackupLocationMap := make(map[string]string)
+	var backupList []string
+	var appContexts []*scheduler.Context
+	var backupLocation string
+	var bkpNamespaces []string
+	var clusterUid string
+	var clusterStatus api.ClusterInfo_StatusInfo_Status
+	bkpNamespaces = make([]string, 0)
+	providers := getProviders()
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_locked_bucket_test.go:37]: done

File: tests/backup/backup_locked_bucket_test.go [hunk 3]
```diff
@@ -0,0 +1,407 @@
...
+					CredName := fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+					bucketName := fmt.Sprintf("%s-%s-%s-locked", provider, getGlobalLockedBucketName(provider), strings.ToLower(mode))
+					backupLocation = fmt.Sprintf("%s-%s-%s-lock", provider, getGlobalLockedBucketName(provider), strings.ToLower(mode))
+					err := CreateS3Bucket(bucketName, true, 3, mode)
+					log.FailOnError(err, "Unable to create locked s3 bucket %s", bucketName)
+					BackupLocationUID = uuid.New()
+					BackupLocationMap[BackupLocationUID] = backupLocation
+					err = CreateBackupLocation(provider, backupLocation, BackupLocationUID, CredName, CloudCredUID,
+						bucketName, orgID, "")
+					dash.VerifyFatal(err, nil, "Creating backup location")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_locked_bucket_test.go:112]: done

File: tests/backup/backup_locked_bucket_test.go [hunk 4]
```diff
@@ -0,0 +1,407 @@
...
+			log.InfoD("Creating backup location for unlocked bucket")
+			for _, provider := range providers {
+				CredName := fmt.Sprintf("%s-%s-%v", "cred", provider, time.Now().Unix())
+				bucketName := fmt.Sprintf("%s-%s", provider, getGlobalBucketName(provider))
+				backupLocation = fmt.Sprintf("%s-%s-unlockedbucket", provider, getGlobalBucketName(provider))
+				BackupLocationUID = uuid.New()
+				BackupLocationMap[BackupLocationUID] = backupLocation
+				err := CreateBackupLocation(provider, backupLocation, BackupLocationUID, CredName, CloudCredUID,
+					bucketName, orgID, "")
+				dash.VerifyFatal(err, nil, "Creating backup location")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_locked_bucket_test.go:128]: done

File: tests/backup/backup_locked_bucket_test.go [hunk 5]
```diff
@@ -0,0 +1,407 @@
...
+			}
+		})
+
+		Step("Register cluster for backup", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			err = CreateSourceAndDestClusters(orgID, "", "", ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+			clusterStatus, clusterUid = Inst().Backup.RegisterBackupCluster(orgID, SourceClusterName, "")
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, "Verifying backup cluster")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_locked_bucket_test.go:138]: done

File: tests/backup/backup_locked_bucket_test.go [hunk 6]
```diff
@@ -0,0 +1,407 @@
...
+				}
+			}
+		})
+		Step("Restoring the backups application", func() {
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for range bkpNamespaces {
+				for _, backupName := range backupList {
+					err = CreateRestore(fmt.Sprintf("%s-restore", backupName), backupName, nil, SourceClusterName, orgID, ctx, make(map[string]string))
+					dash.VerifyFatal(err, nil, "Restore failed")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_locked_bucket_test.go:162]: done

File: tests/backup/backup_locked_bucket_test.go [hunk 7]
```diff
@@ -0,0 +1,407 @@
...
+				for _, mode := range modes {
+					bucketName := fmt.Sprintf("%s-%s-%s", provider, getGlobalLockedBucketName(provider), strings.ToLower(mode))
+					backupLocation = fmt.Sprintf("%s-%s-%s-lock", provider, getGlobalLockedBucketName(provider), strings.ToLower(mode))
+					err := CreateS3Bucket(bucketName, true, 3, mode)
+					log.FailOnError(err, "Unable to create locked s3 bucket %s", bucketName)
+					BackupLocationUID = uuid.New()
+					BackupLocationMap[BackupLocationUID] = backupLocation
+					err = CreateBackupLocation(provider, backupLocation, BackupLocationUID, credName, CloudCredUID,
+						bucketName, orgID, "")
+					dash.VerifyFatal(err, nil, "Creating backup location")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_locked_bucket_test.go:300]: done

File: tests/backup/backup_portworx_test.go [hunk 8]
```diff
@@ -0,0 +1,563 @@
...
+
+		Step("Taking backup of applications", func() {
+			log.InfoD("Taking backup of applications")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, namespace := range bkpNamespaces {
+				backupName = fmt.Sprintf("%s-%s-%v", BackupNamePrefix, namespace, time.Now().Unix())
+				err = CreateBackup(backupName, SourceClusterName, backupLocationName, BackupLocationUID, []string{namespace},
+					nil, orgID, clusterUid, "", "", "", "", ctx)
+				dash.VerifyFatal(err, nil, "Verifying backup creation")
```
[the remark this answers is not in the document]
Comment (snigdha-px) [reply tests/backup/backup_portworx_test.go:83]: done

**Review by lsrinivas-pure — APPROVED** [review]
lgtm

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**